docs/docsite/rst/os_guide/windows_performance.rst

```diff
@@ -20,24 +20,50 @@ seconds.
 
 .. code-block:: powershell
 
-    function Optimize-PowershellAssemblies {
-      # NGEN powershell assembly, improves startup time of powershell by 10x
-      $old_path = $env:path
-      try {
-        $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
-        [AppDomain]::CurrentDomain.GetAssemblies() | % {
-          if (! $_.location) {continue}
-          $Name = Split-Path $_.location -leaf
-          if ($Name.startswith("Microsoft.PowerShell.")) {
-            Write-Progress -Activity "Native Image Installation" -Status "$name"
-            ngen install $_.location | % {"`t$_"}
-          }
-        }
-      } finally {
-        $env:path = $old_path
-      }
+    function Optimize-Assemblies {
+        param (
+            [string]$assemblyFilter = "Microsoft.PowerShell.",
+            [string]$activity = "Native Image Installation"
+        )
+        
+        # Save the current $env:path
+        $old_path = $env:path
+        
+        try {
+            # Set $env:path to the runtime directory
+            $env:path = [Runtime.InteropServices.RuntimeEnvironment]::GetRuntimeDirectory()
+            
+            # Get a list of loaded assemblies
+            $assemblies = [AppDomain]::CurrentDomain.GetAssemblies()
+            
+            # Filter assemblies based on the provided filter
+            $filteredAssemblies = $assemblies | Where-Object { $_.Location -match $assemblyFilter }
+            
+            if ($filteredAssemblies.Count -eq 0) {
+                Write-Host "No matching assemblies found for optimization."
+                return
+            }
```

Comment:
I don't think this is needed personally; the `foreach` will handle the case of no results by not performing any actions.

But I wouldn't block its inclusion either. Instead of `$filteredAssemblies.Count -eq 0` I'd probably use `- not $filteredAssemblies`.

Reply:
@briantist I agree with you mostly. From a standpoint of minimizing code sure, but this is published code I would expect people to copy and paste. I mean I sure did when I found it. We can't expect everyone to know what they are doing or how to read and troubleshoot scripts. From my experience it's best to give people some error statements if things happen that are unexpected.